app/controllers/users_controller.rb

```diff
@@ -0,0 +1,72 @@
+require './config/environment'
+require 'rack-flash'
+
+class UsersController < ApplicationController
+  
+  use Rack::Flash
+  configure do
+    set :public_folder, 'public'
+    set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+   get '/users/:slug' do 
+    @user = User.find_by_slug(params[:slug])
+    
+    erb :'/users/user_homepage'
+  end
+
+  # Signup with a username, login, password
+  # If a session is logged in - it should redirect to the users page
+  get '/signup' do 
+    if logged_in?  # if a user session isnt already live, populate the signup
+      redirect to '/tweets'
+    else
+      erb :'/signup'
+    end
+  end
+
+  # catches the signup params - creates a new user - and logs them into the tweets homepage
+  post '/signup' do 
+    if params.values.any? { |el| el.empty? }
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.